### ISSUE TITLE
Misuse of mBluetoothAdapter.isMultipleAdvertisementSupported()

### ISSUE THREAD

**Issue by farmazon3000** [WebBluetoothCG/ble-test-peripheral-android#67]
In https://github.com/WebBluetoothCG/ble-test-peripheral-android/blob/master/app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java there is:
```
if (!mBluetoothAdapter.isMultipleAdvertisementSupported()) {
  Toast.makeText(this, R.string.bluetoothAdvertisingNotSupported, Toast.LENGTH_LONG).show();
  Log.e(TAG, "Advertising not supported");
}
```

And this is misuse of `isMultipleAdvertisementSupported`, check this https://github.com/AltBeacon/android-beacon-library/issues/266:
```
It is important to understand that just because isMultipleAdvertisementSupported() returns false does not mean that the device cannot advertise at all -- it means either that it cannot advertise at all OR that it can only transmit one advertisement at a time. Many Motorola devices (Moto X, Moto G) behave this way. The way you are supposed to check if advertising is supported at all is to get an instance of BluetoothLEAdvertiser. If the call to get an instance fails, then it is not supported.
```

**Comment by g-ortuno** [comment]
I'm wary of adding code to support one specific vendor. The Android documentation explicitly states that isMultipleAdvertisementSupported() should be used to check for advertising support.

**Comment by farmazon3000** [comment]
`The Android documentation explicitly states that isMultipleAdvertisementSupported() should be used to check for advertising support`

Does it? No it does not.

What it does says is https://developer.android.com/reference/android/bluetooth/BluetoothAdapter.html#isMultipleAdvertisementSupported()
`Return true if the multi advertisement is supported by the chipset`

**Comment by g-ortuno** [comment]
Should have included a citation in my original response. It is under [getBluetoothLeAdvertiser](https://developer.android.com/reference/android/bluetooth/BluetoothAdapter.html#getBluetoothLeAdvertiser()):

> Use isMultipleAdvertisementSupported() to check whether LE Advertising is supported on this device before calling this method.

**Comment by phstudy** [comment]
@g-ortuno I think @farmazon3000 is right.
Just checking the response of `getBluetoothLeAdvertiser()` is null or not, it can determine whether 
the device supports LE advertising. The comment of `getBluetoothLeAdvertiser()` is not accurate, you can check the source of the method. 

[BluetoothAdapter.java in ASOP master](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/bluetooth/BluetoothAdapter.java#662)

[BluetoothAdapter.java in ASOP 7.1_r9](https://android.googlesource.com/platform/frameworks/base/+/android-cts-7.1_r9/core/java/android/bluetooth/BluetoothAdapter.java#573)

**Comment by scheib** [comment]
Ok.  For BLE Test Peripheral, is this check causing you to not be able to use the tool?

**Comment by phstudy** [comment]
@scheib Yes. If I can not pass the checking, BLE Test Peripheral will show `LE Advertising is not available. Please pair with a device to be discoverable.`.

**Comment by AlejandroHCruz** [comment]
Hi, any conclusion on this subject?

**Comment by nowavewater** [comment]
Actually, it's not required to support multiple advertisement for Android device to act as peripheral role. Check the [official example](https://github.com/androidthings/sample-bluetooth-le-gattserver/tree/master/java). 

I have a device which returns false with `isMultipleAdvertisementSupported()`, it works just with the official example.

**Comment by g-ortuno** [comment]
I'm convinced. Feel free to send a CL to change this.

**Comment by abdoutech93** [comment]
> Actually, it's not required to support multiple advertisement for Android device to act as peripheral role. Check the [official example](https://github.com/androidthings/sample-bluetooth-le-gattserver/tree/master/java).
> 
> I have a device which returns false with `isMultipleAdvertisementSupported()`, it works just with the official example.

THANKS A LOT, you made my day
